### PR TITLE
Implement drmkpi kthread.

### PR DESCRIPTION
**Pull request by bukinr**
drmkpi implemented using no task_struct.
Works fine with DRM scheduler and panfrost

### CODE REVIEW

File: drmkpi/drmkpi_kthread.c [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+__FBSDID("$FreeBSD$");
+
+#include <sys/proc.h>
+#include <sys/bus.h>
+#include <sys/interrupt.h>
+#include <sys/priority.h>
+
+#include <linux/kthread.h>
+
+#define	DRMKPI_SUSPEND_WAIT	2000000
```
Comment (evadot) [inline drmkpi/drmkpi_kthread.c:41]: Any reason for this value ?

Comment (bukinr) [reply drmkpi/drmkpi_kthread.c:41]: We need to pass something to kthread_suspend(). But we can pass 0 and wait forever, which is less desirable I think

File: drmkpi/drmkpi_kthread.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+bool
+drmkpi_kthread_should_stop(void)
+{
+	struct thread *td;
+
+	td = curthread;
+
+	if (td->td_flags & TDF_KTH_SUSP)
```
[the remark this answers is not in the document]
Comment (bukinr) [reply drmkpi/drmkpi_kthread.c]: Yes, updated.

**Comment by evadot** [comment]
Looks ok, will merge this weekend unless @strejda have any objection.

**Pull request merged**